src/ds3.c

```diff
@@ -818,8 +855,8 @@ static ds3_str* _build_path(const char* path_prefix, const char* bucket_name, co
 
     path = ds3_str_init(full_path);
 
-    g_free(full_path);
     g_free(joined_path);
+    g_free(full_path);
```

Comment:
Why was this change made?  Not that it should affect anything, but was there a reason?